eclipse-theia/theia
# Cleanup deprecated and out-dated authentication API

#### What it does
- Remove very out-dated backwards compatibility to proposed VS Code API to ensure we can have proper compatible types and no conflicts between the deprecated proposed API and the stable API 
  - Get rid of provider id-based API
  - Remove deprecated 'hasSession' from API
  - Deprecate login/logout (now: createSession/removeSession) in Theia

Previously we were compatible with a very specific commit of the proposed authentication API of VS Code 1.53.2. The API was moved from its proposed state to stable with 1.54.0 to which we are fully compatible even after this commit.

For a list of how the current compatibility came to be, please see https://github.com/eclipse-theia/theia/pull/10709#issuecomment-1029910285

Relates to https://github.com/eclipse-theia/theia/issues/11556

#### How to test
Check out commit https://github.com/martin-fleck-at/theia/tree/issue-1156-test which adds an authentication provider based on the stable API. We have commands that showcase the log in and log out of an account. This is basically an updated version of the testing strategy here: https://github.com/eclipse-theia/theia/pull/10709

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
